### DISCUSSION TITLE
[14.0->15.0] Need help to understand log

### DISCUSSION THREAD

**Discussion by Gabrio2** [OCA/OpenUpgrade#4596] [Q&A] [answered]
Dear, I hope you are fine. 
I cannot figure out from where the following error comes from.

The complete log output, got it from running OpenUpgrade 14to15, is in attach,
but in the text below I extracted the core part of the error. 

I also query into my DB sigmaBEdb.tmp and I checked inside the table hr_employee.
It looks to me like a misleading log message.

Any idea on how to resolve this issue, it would be greatly appreciated.
Gabriele

_2024-10-21 08:17:07,628 119563 ERROR sigmaBEdb.tmp odoo.sql_db: bad query: INSERT INTO "hr_employee" ("id", "active", "address_home_id", "address_id", "certificate", "color", "company_id", "create_date", "create_uid", "department_id", "employee_type", "marital", "name", "resource_calendar_id", "resource_id", "user_id", "work_email", "work_permit_scheduled_activity", "write_date", "write_uid") VALUES (nextval('hr_employee_id_seq'), true, 54, 1, 'other', 0, 1, '2024-10-21 08:17:05.011787', 1, 1, 'employee', 'single', 'Gabriele', 3, 54, 2, 'FAKE-USER@FAKE-COMPANY-NAME.com', false, '2024-10-21 08:17:05.011787', 1) RETURNING id
ERROR: duplicate key value violates unique constraint "hr_employee_user_uniq"
DETAIL:  Key (user_id, company_id)=(2, 1) already exists._

[log_openupdate_sigmaBEdb.tmp.txt](https://github.com/user-attachments/files/17458051/log_openupdate_sigmaBEdb.tmp.txt)

**Comment by hbrunn** [comment, marked as answer]
well pretty much what the error message says: It tries to create an [employee](https://github.com/OCA/OCB/blob/15.0/addons/hr/data/hr_data.xml#L20) with the admin user as user, but you seem to already have such an employee with a different or no xmlid.

Probably you deleted the employee that [v14](https://github.com/OCA/OCB/blob/14.0/addons/hr/data/hr_data.xml#L21) created for the admin user, and then created a new employee with the admin user as user some time afterwards. (same error should happen if you say `-i hr` in you v14 installation)

Easiest way to fix your data is to create an xmlid `hr.employee_admin` pointing to your employee that has the admin user as user.

**Comment by Gabrio2** [comment]
Hello @hbrunn I just wish to thank for your fast response and share my solution inspired from your reply:

with "developer mode" look for the Employee (with name Gabriele) and change its XML ID.

The 2 pictures shows what I did , without write into hr_data.xml,  this modification let me 
to pass through the error and upgrade db odoo from 14 to 15 .

BR
![XML ID before](https://github.com/user-attachments/assets/81bbc728-fc5c-4f05-93a8-ec3c50a376ce) 

![XML ID after modification](https://github.com/user-attachments/assets/afdf912c-0911-429e-8895-8407d85eafb1)
